tests/testDipoleFitter.py

```diff
@@ -0,0 +1,252 @@
+    @var w: width (pixels) of test generated exposures
+    @var h: height (pixels) of test generated exposures
+    @var xc: x coordinate (pixels) of center(s) of input dipole(s)
+    @var yc: y coordinate (pixels) of center(s) of input dipole(s)
+    @var flux: flux(es) of input dipole(s)
+    @var gradientParams: tuple with three parameters for linear background gradient
+    @var offsets: pixel coordinates between lobes of dipoles
+    """
+
+    def __init__(self, xc=np.array([65.3, 24.2]), yc=np.array([38.6, 78.5])):
+        """Initialize the parameters."""
+        np.random.seed(666)
+        self.display = False
+        self.verbose = False
+        self.w, self.h = 100, 100  # size of image
+
+        self.xc = xc  # xcenters of two dipoles in image
+        self.yc = yc  # ycenters of two dipoles
+        self.flux = np.array([2500., 2345.])  # fluxes of pos/neg lobes
+        self.gradientParams = np.array([10., 3., 5.])
+
+        self.offsets = np.array([-2., 2.])  # pixel coord offsets between lobes of dipoles
+        self.testImage = ipUtils.DipoleTestImage(
+            xcenPos=self.xc + self.offsets,
+            ycenPos=self.yc + self.offsets,
+            xcenNeg=self.xc - self.offsets,
+            ycenNeg=self.yc - self.offsets,
+            flux=self.flux, fluxNeg=self.flux,
+            gradientParams=self.gradientParams)
+
+        self.catalog = self.testImage.detectDipoleSources()
+
+
+# First, test the algorithm itself (fitDipole()):
+# Create a simulated diffim (with dipoles) and a linear background gradient in the pre-sub images
+#   then compare the input fluxes/centroids with the fitted results.
+class DipoleFitTest(lsst.utils.tests.TestCase):
+    """A test case for dipole fit algorithm/plugin/task.
+
+    Test the dipole fitting algorithm on two dipoles within an image with
+    simulated noise.
+    """
+
+    def testDipoleAlgorithm(self):
+        """Test the dipole fitting algorithm. Test that the resulting
+        fluxes/centroids are very close to the input values for both
+        dipoles in the image.
+        """
+        params = DipoleFitTestGlobalParams()
+
+        if params.verbose:
+            for s in params.catalog:
+                fp = s.getFootprint()
+                print(fp.getBBox(), fp.getNpix())
+                for pk in fp.getPeaks():
+                    print('FOOTPRINT CENTER:', pk.getIy(), pk.getIx(), pk.getPeakValue())
+
+        offsets = params.offsets
+        testImage = params.testImage
+        for i, s in enumerate(params.catalog):
+            alg = DipoleFitAlgorithm(testImage.diffim, testImage.posImage, testImage.negImage)
+            result = alg.fitDipole(
+                s, rel_weight=1., separateNegParams=False,
+                verbose=params.verbose, display=params.display)
+
+            self.assertClose((result.psfFitPosFlux + abs(result.psfFitNegFlux))/2.,
+                             params.flux[i], rtol=0.02)
```

Comment:
These relative tolerances should all be a class-level parameter. Also, why is one 2% and the others 1%? Can the be tighter than that, given that you know what the input is?